File: Data Structures & Algorithms/majority-element/submission-4.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:

        seenNums = dict()
        result, maxCount = 0, 0

        for num in nums:
            seenNums[num] = 1 + seenNums.get(num, 0)

            if maxCount < seenNums[num]:
                result = num
                maxCount = seenNums[num]

        return result
        

File: Data Structures & Algorithms/subsets-ii/submission-4.java
class Solution {
    public List<List<Integer>> subsetsWithDup(int[] nums) {
        Arrays.sort(nums);
        int prevIndex = 0, index = 0;
        List<List<Integer>> result = new ArrayList<>();
        result.add(new ArrayList<>());

        for(int i = 0; i < nums.length; i++) {
            index = (i >= 1 && nums[i] == nums[i - 1]) ? prevIndex : 0;
            prevIndex = result.size();
            for(int j = index; j < prevIndex; j++) {
                List<Integer> temp = new ArrayList<>(result.get(j));
                temp.add(nums[i]);
                result.add(temp);
            }
        }  

        return result;      
    }
}
